src/__tests__/requirement15.test.js

```diff
@@ -1,24 +1,18 @@
 import React from 'react';
 import { render, screen, waitFor, fireEvent } from '@testing-library/react';
 import App from '../App';
-import * as api from '../services/api';
-import mockedCategoriesResult from '../__mocks__/categories';
-import mockedQueryResult from '../__mocks__/query';
-
-jest.mock('../services/api');
-api.getCategories.mockImplementation(
-  () => Promise.resolve(mockedCategoriesResult),
-);
-api.getProductsFromCategoryAndQuery.mockImplementation(
-  () => Promise.resolve(mockedQueryResult),
-);
+import mockFetch from '../__mocks__/mockFetch';
 
 describe('15 - Mostre quais produtos tem o frete grátis', () => {
   it('Exibe corretmente a informação de frete grátis dos produtos', async () => {
+
+    jest.spyOn(global, 'fetch').mockImplementation((url) => {
+      return mockFetch(url);
+    });
     render(<App />);
-    await waitFor(() => expect(api.getCategories).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
     fireEvent.click(screen.getAllByTestId('category')[0]);
-    await waitFor(() => expect(api.getProductsFromCategoryAndQuery).toHaveBeenCalled());
+    await waitFor(() => expect(global.fetch).toHaveBeenCalled());
```

Comment:
:checkered_flag: :tada: 

```suggestion
    await waitFor(() => expect(global.fetch).toHaveBeenCalledTimes(2));
```